models/constraints.py

```diff
@@ -0,0 +1,172 @@
+from datetime import timedelta
+from typing import List
+from ortools.sat.python import cp_model
+from ortools.sat.python.cp_model import CpModel
+
+from .employees.employee import Employee
+from .employees.employee_status_enum import EmployeeStatusEnum
+from .shifts.shift import Shift
+from .shifts.shifts_enum import ShiftTypesEnum
+
+
+class Constraints:
+
+    def __init__(self, employees: List[Employee], shifts: list[Shift]):
+        self.employees = employees
+        self.cp_model = cp_model.CpModel()
+        self.shifts = shifts
+        self.all_possible_shift_employee_combinations = self._generate_all_possible_shift_employee_combinations(self.employees, self.shifts, self.cp_model)
+
+    @staticmethod
+    def _generate_all_possible_shift_employee_combinations(employees: List[Employee], shifts: list[Shift], cp_model):
+        all_possible_shift_employee_combinations = {}
+        for employee in employees:
+            for shift in shifts:
+                all_possible_shift_employee_combinations[
+                    (employee.id, str(shift.start_date_and_time_of_shift.date()),
+                     str(shift.end_date_and_time_of_shift.date()), shift.type.value)
+                ] = cp_model.NewBoolVar(
+                        f"shift_employee{employee.id}_shift_start_date{str(shift.start_date_and_time_of_shift.date())}_shift_end_date{str(shift.end_date_and_time_of_shift.date())}_shift{shift.type.value}")
+
+        return all_possible_shift_employee_combinations
+
+    # A constraint that ensures that there will be only one employee in each shift per day
+
+    def add_only_one_employee_in_each_shift_constraint(self):
```

Comment:
why is this one not static and accessing `self.employees` and `_generate_all_possible_shift_employee_combinations` isnt?

Reply:
for other funcs here too